core/ordering/cosipbft/blockstore/mem.go

```diff
@@ -170,5 +170,16 @@ type observer struct {
 }
 
 func (obs observer) NotifyCallback(evt interface{}) {
-	obs.ch <- evt.(types.BlockLink)
+	for {
```

Comment:
that feels strange that an observer can actually misses some events because he wasn't quick enough. If that doesn't pose a problem it should be well mentioned.

Reply:
Your feeling is right, that was a dirty solution to fix a blocking issue (like you planned it with the remark on the watcher basically), but I will make a PR to use a non-blocking buffer like mino. Adding a TODO for the moment.